EU4toV2/Source/V2World/Country/Country.h

```diff
@@ -0,0 +1,166 @@
+#include "../State/State.h"
+#include "../Leader/Leader.h"
+#include "../Army/Army.h"
+#include "../Diplomacy/Relation.h"
+
+namespace EU4
+{
+	class Country;
+	class World;
+	class Regions;
+}
+namespace mappers
+{
+	class RegimentCostsMapper;
+	class AdjacencyMapper;
+	class LeaderTraitMapper;
+	class PortProvinces;
+	class Unreleasables;
+	class TechSchoolMapper;
+	class CultureMapper;
+	class ReligionMapper;
+	class GovernmentMapper;
+	class CountryMappings;
+	class PartyNameMapper;
+	class PartyTypeMapper;
+	class IdeaEffectMapper;
+	class TechGroupsMapper;
+}
+namespace V2
+{
+	class Relation;
+	class Country
+	{
+	public:
+		Country() = default;
+		Country(
+			const std::string& countriesFileLine, 
+			bool _dynamicCountry, 
+			const mappers::PartyNameMapper& partyNameMapper, 
+			const mappers::PartyTypeMapper& partyTypeMapper);
+		Country(
+			const std::string& _tag, 
+			const std::string& _commonCountryFile, 
+			const mappers::PartyNameMapper& partyNameMapper,
+			const mappers::PartyTypeMapper& partyTypeMapper);
+		void initFromEU4Country(
+			const EU4::Regions& eu4Regions,
+			std::shared_ptr<EU4::Country> _srcCountry,
+			const mappers::TechSchoolMapper& techSchoolMapper,
+			const mappers::CultureMapper& cultureMapper,
+			const mappers::CultureMapper& slaveCultureMapper,
+			const mappers::IdeaEffectMapper& ideaEffectMapper,
+			const mappers::ReligionMapper& religionMapper,
+			const mappers::ProvinceMapper& provinceMapper,
+			const mappers::GovernmentMapper& governmentMapper,
+			const mappers::CountryMappings& countryMapper
+		);
+		void initFromHistory(const mappers::Unreleasables& unreleasablesMapper);
+		void setupPops(
+			double popWeightRatio,
+			int popConversionAlgorithm,
+			const mappers::ProvinceMapper& provinceMapper
+		);
+		void convertArmies(
+			const mappers::RegimentCostsMapper& regimentCostsMapper,
+			const std::map<int, std::shared_ptr<V2::Province>>& allProvinces,
+			const mappers::PortProvinces& portProvincesMapper,
+			const mappers::ProvinceMapper& provinceMapper
+		);
+		
+		void setNewCountry() { newCountry = true; }
+		bool isNewCountry() const { return newCountry; }
+		std::string getTag() const { return tag; }
+		std::shared_ptr<EU4::Country> getSourceCountry() const { return srcCountry; }
+		std::shared_ptr<Country> getColonyOverlord() const { return colonyOverlord; }
+		void setColonyOverlord(std::shared_ptr<Country> colony) { colonyOverlord = colony; }
+		int getNumFactories() const { return details.numFactories; }
+		Relation& getRelation(const std::string& withWhom);
+		void addRelation(Relation& newRelation);
+		void absorbVassal(std::shared_ptr<Country> vassal);
+		const std::map<int, std::shared_ptr<Province>>& getProvinces() const { return provinces; }
+		void addPrestige(double additionalPrestige) { details.prestige += additionalPrestige; }
+		double getPrestige() const { return details.prestige; }
+		std::tuple<double, double, double> getNationalValueScores();
```

Comment:
thank you for pointing this out. Entire function was wrong as it returned original scrcountry scores instead of calculated ones. all investments are now broken into structs.